general/mem_manager.hpp

```diff
@@ -32,13 +32,15 @@ enum class MemoryType
    HOST_64,        ///< Host memory; aligned at 64 bytes
    HOST_DEBUG,     ///< Host memory; allocated from a "host-debug" pool
    HOST_UMPIRE,    ///< Host memory; using Umpire
+   HOST_PINNED,    ///< Host memory: pinned (page-locked)
    MANAGED,        /**< Managed memory; using CUDA or HIP *MallocManaged
                         and *Free */
    DEVICE,         ///< Device memory; using CUDA or HIP *Malloc and *Free
    DEVICE_DEBUG,   /**< Pseudo-device memory; allocated on host from a
                         "device-debug" pool */
    DEVICE_UMPIRE,  ///< Device memory; using Umpire
-   SIZE            ///< Number of host and device memory types
+   SIZE,           ///< Number of host and device memory types
+   PRESERVE        ///< Default parameter type; preserves existing behavior
```

Comment:
Not sure that's the right place to put this enum - `KEEP` vs `PRESERVE` ?

Reply:
This was requested for cases like [this](https://github.com/mfem/mfem/blob/feature/tomstitt/temp-mem-type/linalg/sparsemat.hpp#L150-L151), to make it clear what the `mt` arg was doing in the default case